src/storage/txn/scheduler.rs

```diff
@@ -481,6 +546,46 @@ impl<E: Engine, L: LockMgr> MsgScheduler for Scheduler<E, L> {
             _ => unreachable!(),
         }
     }
+
+    fn on_batch_msg(&self, tasks: Vec<(u64, Msg)>) {
+        if tasks.is_empty() {
+            return;
+        }
+        let cid = match tasks[0].1 {
+            Msg::ReadFinished { cid, .. } => cid,
+            Msg::WriteFinished { cid, .. } => cid,
+            Msg::FinishedWithErr { cid, .. } => cid,
+            _ => panic!("msg type mismatch"),
+        };
+        self.on_batch_finish(
+            cid,
+            tasks
+                .into_iter()
+                .map(|(id, msg)| {
+                    (
+                        id,
+                        match msg {
+                            Msg::ReadFinished { pr, .. } => pr,
+                            Msg::WriteFinished { pr, result, .. } => match result {
+                                Ok(()) => pr,
+                                Err(e) => ProcessResult::Failed {
+                                    err: StorageError::from(e),
+                                },
+                            },
+                            Msg::FinishedWithErr { err, .. } => ProcessResult::Failed {
+                                err: StorageError::from(err),
+                            },
+                            _ => unreachable!(),
+                        },
+                    )
+                })
+                .collect(),
+        );
+    }
+
+    fn on_batch_finished(&self, cid: u64) {
```

Comment:
It looks like similar to `on_batch_finish`.  Could it be replaced with another name?